src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -261,31 +268,75 @@ internal override async Task<CompletionChange> GetChangeAsync(Document document,
                 return CompletionChange.Create(Utilities.Collapse(newText, builder.ToImmutableAndFree()));
             }
 
-            static bool ShouldCompleteWithFullyQualifyTypeName(Document document)
+            // Apply the change of fully qualified type name, then try to simplify it.
+            static async Task<TextChange> GetSimplifiedChange(Document document, TextChange fullyQualifiedChange, CancellationToken cancellationToken)
+            {
+                var originalTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                var originalText = await originalTree.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+                var fullyQualifiedText = originalText.WithChanges(fullyQualifiedChange);
+                var fullyQualifiedTree = originalTree.WithChangedText(fullyQualifiedText);
+                var fullyQualifiedRoot = await fullyQualifiedTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+                var fullyQualifiedSpan = new TextSpan(fullyQualifiedChange.Span.Start, fullyQualifiedChange.NewText.Length);
+                var fullyQualifiedNode = fullyQualifiedRoot.FindNode(fullyQualifiedSpan);
+                var fullyQualifiedNodeWithAnnotation = fullyQualifiedNode.WithAdditionalAnnotations(Simplifier.Annotation);
+
+                var annotatedRoot = fullyQualifiedRoot.ReplaceNode(fullyQualifiedNode, fullyQualifiedNodeWithAnnotation);
+                var annotatedDocument = document.WithSyntaxRoot(annotatedRoot);
+
+                var simplifiedDocument = await Simplifier.ReduceAsync(annotatedDocument, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                var changes = await simplifiedDocument.GetTextChangesAsync(document, cancellationToken).ConfigureAwait(false);
+                return Utilities.Collapse(originalText, changes.ToImmutableArray());
```

Comment:
this is a surprisingly large amount of work to have to do.

Reply:
Yea, I'm wondering if there's a better way to archive this.